libr/anal/anal.c

```diff
@@ -265,7 +265,7 @@ R_API bool r_anal_use(RAnal *anal, const char *name) {
 	return false;
 }
 
-R_API char *r_anal_get_reg_profile(RAnal *anal) {
+R_API RStrBuf *r_anal_get_reg_profile(RAnal *anal) {
```

Comment:
why. whats the benefit of this.strbuf is useful when the string is suposed to be resized and modified many times, this is not the case for the regprofile, so i dont see the benefit

Reply:
We can use RStrBuf for storing pointers to constant strings. See the PR description.

Reply:
strbuf changes the pointer depending of the length of the string, also, what about returning const char * and int *len? and then just delegate the logic to compare pointer in a single place

Reply:
> strbuf changes the pointer depending of the length of the string
Can you explain more?

Reply:
to store pointers to constant strings we may probably want to use RStrConst instead, but this api changed recently to be thread safe so its not an option here anymore.